systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java

```diff
@@ -986,4 +1012,473 @@ public void testWatchingOtherNamespace() throws InterruptedException {
+        waitForJobSuccess(readMessagesFromClusterJob(kafkaSourceName, nameConsumerSource, topicSourceName, messagesCount, null, false));
+        // Create job to read 20 records using Kafka consumer for target cluster
+        Job jobReadMessagesForTarget = waitForJobSuccess(readMessagesFromClusterJob(kafkaTargetName, nameConsumerTarget, topicSourceName, messagesCount, null, false));
+        // Check consumed messages in target cluster
+        checkRecordsForConsumer(messagesCount, jobReadMessagesForTarget);
+    }
+
+    /**
+     * Test mirroring messages by Mirror Maker over tls transport using mutual tls auth
+     */
+    @Test
+    @JUnitGroup(name = "regression")
+    public void testMirrorMakerTlsAuthenticated() {
+        String topicSourceName = TOPIC_NAME + "-source" + "-" + rng.nextInt(Integer.MAX_VALUE);
+        String nameProducerSource = "send-messages-producer-source";
+        String nameConsumerSource = "send-messages-consumer-source";
+        String nameConsumerTarget = "send-messages-consumer-target";
+        String kafkaUser = "my-user";
+        String kafkaSourceName = CLUSTER_NAME + "-source";
+        String kafkaTargetName = CLUSTER_NAME + "-target";
+        int messagesCount = 20;
+
+        KafkaListenerAuthenticationTls auth = new KafkaListenerAuthenticationTls();
+        KafkaListenerTls listenerTls = new KafkaListenerTls();
+        listenerTls.setAuth(auth);
+
+        // Deploy source kafka with tls listener and mutual tls auth
+        resources().kafka(resources().defaultKafka(CLUSTER_NAME + "-source", 3)
+                .editSpec()
+                    .editKafka()
+                        .withNewListeners()
+                            .withTls(listenerTls)
+                            .withNewTls()
+                            .endTls()
+                        .endListeners()
+                    .endKafka()
+                .endSpec().build()).done();
+
+        // Deploy target kafka with tls listener and mutual tls auth
+        resources().kafka(resources().defaultKafka(CLUSTER_NAME + "-target", 3)
+                .editSpec()
+                    .editKafka()
+                        .withNewListeners()
+                            .withTls(listenerTls)
+                            .withNewTls()
+                            .endTls()
+                        .endListeners()
+                    .endKafka()
+                .endSpec().build()).done();
+
+        // Deploy topic
+        resources().topic(kafkaSourceName, topicSourceName).done();
+
+        // Create Kafka user
+        KafkaUser user = resources().tlsUser(kafkaUser).done();
+        waitTillSecretExists(kafkaUser);
+
+        // Initialize CertSecretSource with certificate and secret names for consumer
+        CertSecretSource certSecretSource = new CertSecretSource();
+        certSecretSource.setCertificate("ca.crt");
+        certSecretSource.setSecretName(kafkaSourceName + "-cluster-ca-cert");
```

Comment:
> -cluster-ca-cert

5 times in the file, maybe create constant for that as well?

Reply:
Yeah, it looks funny, I created the method `clusterCaCertSecretName()` but forgot to update code everywhere. @Frawless @ppatierno thanks. Will fix ti